File: RB101/small_problems/easy_9/welcome_stranger.rb
# Input: An array with a name separated and a hash with title and occupation
# Output: a greeting using the full name, their title and occupation

# Algo
# Pretty straight forward
# join the strings in the array to form a name string
# use string interpolation and print a greeting inserting name
# And the key values from the hash

def greetings(name, info)
  name = name.join(' ')
  title = info[:title]
  job = info[:occupation]
  puts "=> Hello, #{name}! Nice to have a #{title} #{job} around."
end

# Test case
greetings(['John', 'Q', 'Doe'], { title: 'Master', occupation: 'Plumber' })
# => Hello, John Q Doe! Nice to have a Master Plumber around.

File: PY101/small_problems/easy_2/welcome_stranger.py
# Create a function that takes 2 arguments, a list and a dictionary. The list will contain 2 or more elements that, when joined with spaces, will produce a person's name. The dictionary will contain two keys, "title" and "occupation", and the appropriate values. Your function should return a greeting that uses the person's full name, and mentions the person's title.

def greetings(name_list, info_dict):
    full_name = ' '.join(name_list)
    title = info_dict.get('title')
    job = info_dict.get('occupation')

    return f'Hello, {full_name}! Nice to have a {title} {job} around.'

greeting = greetings(
    ["John", "Q", "Doe"],
    {"title": "Master", "occupation": "Plumber"},
)
print(greeting)
# Hello, John Q Doe! Nice to have a Master Plumber around.